Sprint-1/fix/median.js

```diff
@@ -2,10 +2,21 @@
 // Start by running the tests for this function
 // If you're in the Sprint-1 directory, you can run `npm test -- fix` to run the tests in the fix directory
 
-function calculateMedian(list) {
-  const middleIndex = Math.floor(list.length / 2);
-  const median = list.splice(middleIndex, 1)[0];
-  return median;
+const calculateMedian = (list) => {
+  list.sort((a, b) => {
+    return a - b;
+  });
+
+  if (list.length % 2 === 0) {
+    const middleIndex = list.length / 2;
+    console.log((list[middleIndex - 1] + list[middleIndex]) / 2);
+    return (list[middleIndex - 1] + list[middleIndex]) / 2;
+  }
+  else {
+    return list[Math.floor(list.length / 2)];
```

Comment:
The value of `Math.floor(list.length / 2)` is suitable for both odd and even length array. So you could possibly just calculate the middle index once in your code.